src/lsp-error-reporter.cpp

```diff
@@ -102,7 +102,10 @@ void lsp_error_formatter::write_before_message(severity sev,
   this->output_.append_decimal_integer(r.end.line);
   this->output_.append_copy(u8",\"character\":");
   this->output_.append_decimal_integer(r.end.character);
-  this->output_.append_copy(u8"}},\"severity\":1,\"message\":\"");
+  if (sev == severity::error)
+    this->output_.append_copy(u8"}},\"severity\":1,\"message\":\"");
+  else if(sev == severity::warning)
+    this->output_.append_copy(u8"}},\"severity\":2,\"message\":\"");
```

Comment:
* If we add a new severity type, this code could output corrupt JSON.
  * Possible solution: A `switch` statement would ensure we check each severity. If we add a new severity, the compiler will complain about a missing `case`.
* Nit: A lot of the appended string is duplicated in the source code. Remove the duplication if it's reasonable to do so.